torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -261,12 +269,16 @@ void RequestCallbackImpl::processRpc(
             [responseFuture, messageId, rref](
                 const rpc::Message& /* unused */,
                 const c10::optional<utils::FutureError>& error) {
-              if (!error) {
+              if (error) {
+                responseFuture->setError(error->what());
+                return;
+              }
+              try {
                 Message m = ScriptRRefFetchRet({rref->getValue()}).toMessage();
                 m.setId(messageId);
                 responseFuture->markCompleted(std::move(m));
```

Comment:
Did we add these try-catch blocks here because there was something that used to throw and we weren't catching it here?

Reply:
No, we're propagating the exception in OwnerRRef now.